docs/readmes/bazel/bazel.md

```diff
@@ -0,0 +1,510 @@
+
+Usually these commands should not be needed, however they can be used to reclaim disk space or to recover a consistent state when debugging.
+
+> Info: The `bazel clean` command does **not** remove the disk caches. To remove the disk caches the cache folders, e.g. `$MAGMA_ROOT/.bazel-cache` and `$MAGMA_ROOT/.bazel-cache-repo`, need to be removed manually. The exact locations of the disk caches may depend on the environment.
+
+## Testing
+
+There are four categories of tests that concern the AGW, each of these categories is covered in the following sections:
+
+- [unit tests](#unit-tests)
+- [LTE integration tests](#lte-integration-tests)
+- [Python sudo tests](#python-sudo-tests)
+- [load tests](#load-tests)
+
+> Warning: Note that not all bazel test targets in the Magma AGW can be executed directly with bazel. These special targets are tagged in their attributes and the following sections explain how they can be run.
+
+### Unit tests
+
+See the [notes on unit testing](../lte/dev_unit_testing.md) for more detailed information and examples on unit testing.
+
+The unit tests can be run on any of the Docker [environments](#environments) or on the magma-dev and magma-deb VMs.
+
+To run unit tests with bazel, use the command `bazel test` and then specify the [bazel targets](#targets) using the same [command line syntax](#target-command-line-syntax) described in the build section.
+
+For example, to run all unit tests, that are not tagged as manual, run `bazel test //...` and to run a specific unit test specify the test target directly, e.g. `bazel test //lte/gateway/c/core/oai/test/pipelined_client:pipelined_client_test`.
+
+Some useful command line options for unit testing include:
+
+- `--cache_test_results=no` disables the caching of test results and forces a re-run even if there are no changes.
+- `--runs_per_test=<integer>` executes all requested test targets exactly `<integer>`-many times, even if runs are successful. This flag can be used e.g. to detect flaky tests.
+- `--flaky_test_attempts=<integer>` executes all test targets that have the `flaky = True` attribute set, up to `<integer>`-many times. If there are failures the test is marked as flaky in the results. Test targets with the attribute `flaky = False` (default) are unaffected.
+
+A complete list of [command line arguments](https://bazel.build/reference/command-line-reference) can be found in the official bazel documentation.
+
+### LTE integration tests
+
+Information on how to run the LTE integration tests with bazel is documented on the [s1ap tests](../lte/s1ap_tests.md) page.
+
+> **Warning**: Do not execute bazel commands as sudo/root user, because it can break your local bazel setup!
```

Comment:
This warning has no context ... maybe: If you try to execute an integration test with `bazel test` you might get a permission error. Do not execute Bazel commands as sudo/root ...